packages/nuxt/src/pages/module.ts

```diff
@@ -185,41 +185,71 @@ export default defineNuxtModule({
     }
 
     if (useExperimentalTypedPages) {
-      const declarationFile = './types/typed-router.d.ts'
+      const addedPagePaths = new Map<string, { file: NuxtPage['file'], name: EditableTreeNode['name'] }>()
+
+      const typedRouterDtsPathRelative = 'types/typed-router.d.ts'
+      const typedRouterDtsPathAbsolute = resolve(nuxt.options.buildDir, typedRouterDtsPathRelative)
+      const readTypedRouterDts = async () => await readFile(typedRouterDtsPathAbsolute, 'utf-8')
+
+      const fileRouteNameMapDtsPathRelative = 'types/typed-pages.d.ts'
+      const fileRouteNameMapDtsPathAbsolute = resolve(nuxt.options.buildDir, fileRouteNameMapDtsPathRelative)
 
       const typedRouterOptions: TypedRouterOptions = {
         routesFolder: [],
-        dts: resolve(nuxt.options.buildDir, declarationFile),
+        dts: typedRouterDtsPathAbsolute,
         logs: nuxt.options.debug && nuxt.options.debug.router,
         async beforeWriteFiles (rootPage) {
           rootPage.children.forEach(child => child.delete())
+
           const pages = nuxt.apps.default?.pages || await resolvePagesRoutes(options.pattern, nuxt)
+          const augmentPagesContext = nuxt._augmentPagesContext ?? {}
+
           if (nuxt.apps.default) {
             nuxt.apps.default.pages = pages
           }
-          const addedPagePaths = new Set<string>()
+
+          addedPagePaths.clear()
```

Comment:
Probably wrong timing, might have caused the missing routes in the file path vs. route name map